File: 1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.py
class Solution:
    def chalkReplacer(self, chalk: List[int], k: int) -> int:
        total_sum = sum(chalk)
        k %= total_sum
        
        for i, c in enumerate(chalk):
            if k < c:
                return i
            k -= c
            
        return 0

File: LeetCode/1894-find-the-student-that-will-replace-the-chalk/1894-find-the-student-that-will-replace-the-chalk.java
class Solution {
    public int chalkReplacer(int[] chalk, int k) {
        long sum = 0;
        for (int c : chalk) {
            sum += c;
        }
        
        k = (int)(k % sum);
        
        for (int i = 0; i < chalk.length; i++) {
            if (k < chalk[i]) {
                return i;
            }
            k -= chalk[i];
        }
        return 0;
    }
}